src/util/msvc.h

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#ifndef UTIL_MSVC_COMPILER_H
+#define UTIL_MSVC_COMPILER_H
+
+#ifdef _MSC_VER
+#if __cplusplus >= 201700L  // with C++17 it's this deprecated
+                            // struct not available anymore
+#   include "util/xcs_functional.hpp"
```

Comment:
... and missing header file "util/xcs_functional.hpp" :-(
In C++11 the std:: binary_function and std::unary_function are deprecated, but in C++17 this structs are removed (like the standard says). I need this additional definitions to avoid this problem at least with msvc, but maybe this should be solved for all targets. I used a proposal from Stackoverflow as a workaround. Maybe it is clearer if I add the file ;-)

Reply:
Don't just copy'n'paste code from Stackoverflow without thinking. Stackoverflow is mostly filled with bad code samples.